### PR TITLE
[Forge] Use Poetry to manage deps

### PR DESCRIPTION
**Pull request by sionescu**
### Description

Add Poetry config and lockfile.

### Test Plan

Doesn't need testing as such because adding a Poetry config file will have no effect until switching the workflows to Poetry instead of Pip.
We need to separate adding the Poetry config from using it because Forge runs workflows from the main branch and will fail on a PR that contains both changes.

Manually tested with:

$ cd testsuite && poetry install
$ ./forge.py

### CODE REVIEW

**Review by perryjrandall — APPROVED** [review]

File: testsuite/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[tool.poetry]
+name = "aptos_forge_cli"
+version = "0.1.0"
+description = "CLI for the Forge E2E testing environment"
+authors = []
+readme = "README.md"
+
+[tool.poetry.dependencies]
+python = "^3.11"
```
[the remark this answers is not in the document]
Comment (sionescu) [reply testsuite/pyproject.toml:9]: Not sure, I think Poetry does that by default.

**Comment by perryjrandall** [comment]
Can you include the command you ran

Also does this work with the workflow?

https://github.com/aptos-labs/aptos-core/blob/main/.github/workflows/workflow-run-forge.yaml#L106-L108

File: testsuite/pyproject.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (geekflyer) [reply testsuite/pyproject.toml:9]: the lockfile has the hard pins

**Comment by geekflyer** [comment]
can you delete the requirements.txt along with this PR?

**Comment by sionescu** [comment]
> can you delete the requirements.txt along with this PR?

I'm not sure what you mean.

**Comment by geekflyer** [comment]
delete this thing https://github.com/aptos-labs/aptos-core/blob/main/testsuite/requirements.txt

**Comment by sionescu** [comment]
@geekflyer done

**Comment by sionescu** [comment]
@geekflyer Deleting requirements.txt is breaking some workflows. I'll do that in the next PR which switches the workflows to Poetry.

**Review by geekflyer — APPROVED** [review]

**Pull request merged**